### PR TITLE
Docs: updates to contributing, collaborator, translation

### PR DESCRIPTION
**Pull request by Ve33y**
## Description
Moved  "collaborating"-specific snippets and sections from CONTRIBUTING to COLLABORATOR_GUIDE. Added an update to how to add translations. (TRANSLATIONS needs a double check to make sure it's the expected update since it was not part of the issue but a slack thread)

## Related Issues
https://github.com/nodejs/nodejs.org/issues/5413

### CODE REVIEW

File: TRANSLATION.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ Fill the language object with the following fields:
 | `hrefLang`   | The language code in [ISO 639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) format         | `fr`         |
 | `enabled`    | If the language is enabled or not                                                                      | `true`       |
 
+Please also add the new locale file to the locales folder `/i18n/locales` and import it in the `/i18n/locales/index.mjs` file.
```
Comment (ovflowd) [inline TRANSLATION.md:43]:
```suggestion
Please also add the new locale file to the locales folder `/i18n/locales` and import it in the `/i18n/locales/index.mjs` file.

```

**Review by ovflowd — APPROVED** [review]
LGTM! Thank you, @Ve33y! Really appreciate it! I left a tiny code suggestion to add an empty line before the next heading 👀

File: COLLABORATOR_GUIDE.md [hunk 2]
```diff
@@ -111,3 +118,112 @@ layout: contribute.hbs
...
 
 See [TRANSLATION.md](./TRANSLATION.md) for the website translation policy.
+
+## Creating Components
+
+The Node.js Website uses **React.js** as a Frontend Library for the development of the Website. React allows us to create user interfaces with a modern take on Web Development.
+
+If you're unfamiliar with React or Web Development in general, we encourage a read before taking on complex issues and tasks as this repository is **not for educational purposes** and we expect you to have a basic understanding of the technologies used.
+
+We also recommend getting familiar with technologies such as [Next.js][], [MDX][], [SCSS][] and "concepts" such as "CSS Modules" and "CSS-in-JS".
```
Comment (mikeesto) [inline COLLABORATOR_GUIDE.md:128]: Could you please fix the reference style links here?

They can likely be removed from https://github.com/nodejs/nodejs.org/pull/5416/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R206-R208 if they're not used elsewhere and added to the bottom of this file instead

File: COLLABORATOR_GUIDE.md [hunk 3]
```diff
@@ -111,3 +118,112 @@ layout: contribute.hbs
...
+  - When using `children` as a prop, use the `FC<PropsWithChildren<MyComponentProps>>` type instead
+  - Alterenatively you can define your type as `type MyComponentProps = PropsWithChildren<{ my other props}>`
+- Each Props type should be prefixed by the name of the Component
+- Components should always be the `default` export of a React Component file
+- Avoid using DOM/Web APIs/`document`/`window` API access within a React Component. Use utilities or Hooks when you need a Reactive state
+- Avoid making your Component too big. Deconstruct it into smaller Components/Hooks whenever possible
+
+## Unit Tests and Storybooks
+
+Each new feature or bug fix should be accompanied by a unit test (when deemed valuable). We use [Jest][] as our test runner and [React Testing Library][] for our React unit tests.
```
Comment (mikeesto) [inline COLLABORATOR_GUIDE.md:181]: Some more reference style links here & in the paragraph below that need to be wired up

**Review by mikeesto — CHANGES_REQUESTED** [review]

**Comment by Ve33y** [comment]
Closing this one in favor of https://github.com/nodejs/nodejs.org/pull/5420

File: COLLABORATOR_GUIDE.md (hunk 2 above)
Comment (Ve33y) [reply COLLABORATOR_GUIDE.md:128]: Thank you @mikeesto - closed this one in favor of https://github.com/nodejs/nodejs.org/pull/5420 changes should be reflected there!

File: COLLABORATOR_GUIDE.md (hunk 3 above)
Comment (Ve33y) [reply COLLABORATOR_GUIDE.md:181]: same as above - thanks once more @mikeesto

**Pull request closed without merging**